UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneExporter.cs

```diff
@@ -96,21 +125,175 @@ public void SaveGLTFandBin(string path, string fileName)
+                        exportTexture(image, outputPath);
+                        break;
+                }
+            }
+        }
+
+        private void exportMetallicGlossTexture(Texture2D texture, string outputPath)
+        {
+            var textureWidth = texture.width;
+            var textureHeight = texture.height;
+            Color[] outputColors = new Color[textureWidth * textureHeight];
+            for (int i = 0; i < outputColors.Length; ++i)
+            {
+                outputColors[i] = new Color(1.0f, 1.0f, 1.0f);
+            }
+            addTexturePixels(ref texture, ref outputColors, IMAGETYPE.R, IMAGETYPE.B);
+            addTexturePixels(ref texture, ref outputColors, IMAGETYPE.A, IMAGETYPE.G_INVERT);
+            Texture2D newtex = new Texture2D(textureWidth, textureHeight);
+            newtex.SetPixels(outputColors);
+            newtex.Apply();
+            File.WriteAllBytes(Path.Combine(outputPath, texture.name + ".png"), newtex.EncodeToPNG());
+        }
+
+        private void exportTexture(Texture2D texture, string outputPath)
+        {
+            var textureWidth = texture.width;
+            var textureHeight = texture.height;
+            Color[] textureColors = new Color[textureWidth * textureHeight];
+            if (!getPixelsFromTexture(ref texture, out textureColors))
+            {
+                Debug.Log("Failed to convert texture " + texture.name + " (unsupported type or format)");
+                return;
+            }
+            Texture2D newtex = new Texture2D(textureWidth, textureHeight);
+            newtex.SetPixels(textureColors);
+            newtex.Apply();
+            File.WriteAllBytes(Path.Combine(outputPath, texture.name + ".png"), newtex.EncodeToPNG());
+        }
+
+        private void addTexturePixels(ref Texture2D texture, ref Color[] outputColors, IMAGETYPE inputChannel, IMAGETYPE outputChannel)
+        {
+            int textureHeight = texture.height;
+            int textureWidth = texture.width;
+
+            Color[] inputColors = new Color[textureWidth * textureHeight];
+            if (!texture || !getPixelsFromTexture(ref texture, out inputColors))
+            {
+                return;
+            }
+
+            if (textureHeight * textureWidth != outputColors.Length)
+            {
+                Debug.Log("Issue with texture dimensions");
+                return;
+            }
+
+            for (int i = 0; i < textureHeight; ++i)
+            {
+                for (int j = 0; j < textureWidth; ++j)
+                {
+                    int index = i * textureWidth + j;
+                    Color outputColor = outputColors[index];
+                    Color inputColor = inputColors[index];
+                    
+                    switch (inputChannel)
+                    {
+                        case IMAGETYPE.R:
+                            inputColor = new Color(inputColor.r, inputColor.r, inputColor.r, inputColor.r);
```

Comment:
Why are you creating a new Vec4 out of a single float field?

Reply:
This function has been removed.